### PR TITLE
[RF] Add tests ROOT-4580 and ROOT-9528.

### PR DESCRIPTION
**Pull request by hageboeck**
In checking old bug reports, tests for two bugs were written. Although
the bugs had been fixed as side effects of modernisations or other bug
fixes, having those tests in place won't do harm.

### CODE REVIEW

File: roofit/roofitcore/test/testRooDataSet.cxx [hunk 1]
```diff
@@ -44,3 +45,87 @@ TEST(RooDataSet, ImportFromTreeWithCut)
...
+  EXPECT_EQ(static_cast<RooRealVar*>(ds.get(1)->find(*v))->getVal(), 4.);
+  EXPECT_EQ(ds.numEntries(), 2);
+  EXPECT_DOUBLE_EQ(ds.sumEntries("HLT_mu6_mu4_bBmumux_BsmumuPhi_delayed_L1BPH_2M8_MU6MU4_BPH_0DR15_MU6MU4 > 3."), 1.);
+}
+
+
+
+#include <TRandom3.h>
+#include <TH1F.h>
+#include <TCut.h>
```
Comment (eguiraud) [inline roofit/roofitcore/test/testRooDataSet.cxx:74]: includes in the middle of the file are against ROOT coding convention afaik

Comment (hageboeck) [reply roofit/roofitcore/test/testRooDataSet.cxx:74]: Should read that maybe. ;-)

**Comment by hageboeck** [comment]
Oh, these diff-tests ...

**Comment by hageboeck** [comment]
@phsft-bot build just on ROOT-performance-centos7-multicore/default

Updated the failing test roottest:#509, so just checking on one machine if that worked.

**Pull request merged**